.github/workflows/ci.yml

```diff
@@ -57,5 +58,20 @@ jobs:
               run: |
                   pip install -U -e .[tests]
 
+            - name: Set jupyter token env
+              run: echo "JUPYTER_TOKEN=$(openssl rand -hex 32)" >> $GITHUB_ENV
+
+            - name: Install Firefox geckodriver
```

Comment:
I don't understand, here you use this action, but also install it manually below? If it is neccessary to do like this, can you please add a comment with an explanation?